### PR TITLE
Add pxtarget.compile.shimRenames

### PR DESCRIPTION
**Pull request by mmoskal**

### CODE REVIEW

File: pxtcompiler/emitter/emitter.ts [hunk 1]
```diff
@@ -2134,6 +2134,9 @@ ${lbl}: .short 0xffff
                 return emitExpr(args[0])
             }
 
+            if (opts.target.shimRenames && U.lookup(opts.target.shimRenames, nm))
+                nm = opts.target.shimRenames[nm]
```
Comment (pelikhan) [inline pxtcompiler/emitter/emitter.ts:2138]: nit, all in 1 shot?

    nm = opts.target?.shimRenames[nm] || nm

Comment (mmoskal) [reply pxtcompiler/emitter/emitter.ts:2138]: I think that would protect against `opts.target` being `null`, not `opts.target.shimRenames`

Comment (mmoskal) [reply pxtcompiler/emitter/emitter.ts:2138]: I guess there's `opts.target.shimRenames.?[nm]`...

Comment (mmoskal) [reply pxtcompiler/emitter/emitter.ts:2138]: or something

**Pull request merged**